battlebots/database/access.py

```diff
@@ -37,7 +37,8 @@ def remove_bot(user, botname):
                         % (user.nickname, botname, code_dir))
         pass
 
-    bot = session.query(Bot).filter_by(user=user, name=botname).one_or_none()
```

Comment:
(1) is no option as I tried yesterday for the Flask app: the view would try to use an object queried in a "low level" Session which has already been closed, which detaches the object and throws some ObjectDetachedError.